File: src/main/java/com/back/nbe9112team06/domain/meeting/entity/MeetingsDate.java
package com.back.nbe9112team06.domain.meeting.entity;

import com.back.nbe9112team06.global.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Getter
@NoArgsConstructor
public class MeetingsDate extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "meeting_id")
    private Meeting meeting;

    private LocalDate date;

    @Column(name = "created_by")
    private String createdBy;

    @Column(name = "modified_by")
    private String modifiedBy;

    public static MeetingsDate create(LocalDate date, String createdBy) {
        MeetingsDate meetingsDate = new MeetingsDate();
        meetingsDate.date = date;
        meetingsDate.createdBy = createdBy;
        meetingsDate.modifiedBy = createdBy;
        return meetingsDate;
    }

    public void assignMeeting(Meeting meeting) {
        this.meeting = meeting;
    }
}

File: src/main/java/com/back/nbe9112team06/domain/meeting/entity/MeetingsDate.kt
package com.back.nbe9112team06.domain.meeting.entity

import com.back.nbe9112team06.global.entity.BaseEntity
import jakarta.persistence.*
import java.time.LocalDate

@Entity
class MeetingsDate() : BaseEntity() {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "meeting_id")
    var meeting: Meeting? = null
        protected set

    var date: LocalDate? = null
        protected set

    @Column(name = "created_by")
    var createdBy: String? = null
        protected set

    @Column(name = "modified_by")
    var modifiedBy: String? = null
        protected set

    fun assignMeeting(meeting: Meeting) {
        this.meeting = meeting
    }

    constructor(date: LocalDate, createdBy: String) : this() {
        this.date = date
        this.createdBy = createdBy
        this.modifiedBy = createdBy
    }
}
